src/lib/pickles/verify.ml

```diff
@@ -148,31 +148,49 @@ let verify_heterogenous (ts : Instance.t list) =
           in
           (absorb sponge, squeeze)
         in
-        ( absorb evals.ft_eval1 ;
-          let xs = Plonk_types.Evals.to_absorption_sequence evals.evals.evals in
-          let x1, x2 = evals.evals.public_input in
-          absorb x1 ;
-          absorb x2 ;
-          List.iter xs ~f:(fun (x1, x2) ->
-              Array.iter ~f:absorb x1 ; Array.iter ~f:absorb x2 ) ) ;
+        let old_bulletproof_challenges =
+          Vector.map ~f:Ipa.Step.compute_challenges
+            statement.pass_through.old_bulletproof_challenges
+        in
```

Comment:
my understanding of this ^ is that we convert each challenge in the old_bulletproof_challenges (not sure what the type is, it seems to be generic) into another scalar using the endomorphism

Reply:
(also, I'm not sure I understand the need to use endomorphism here, might be good to have an explainer on that too)